app/src/components/ChatBot/chat-bot.tsx

```diff
@@ -0,0 +1,248 @@
+"use client";
+
+import { useCopyToClipboard } from "@/hooks/use-copy-to-clipboard";
+import {
+  Box,
+  Button,
+  Divider,
+  HStack,
+  Icon,
+  IconButton,
+  Spacer,
+  Text,
+  Textarea,
+} from "@chakra-ui/react";
+import { useChat } from "ai/react";
+import { TFunction } from "i18next";
+import { BsStars } from "react-icons/bs";
+import {
+  MdCheckCircle,
+  MdContentCopy,
+  MdOutlineSend,
+  MdOutlineThumbDown,
+  MdOutlineThumbUp,
+  MdRefresh,
+} from "react-icons/md";
+import { ScrollAnchor } from "./scroll-anchor";
+import { RefObject, useRef } from "react";
+
+function useEnterSubmit(): {
+  formRef: RefObject<HTMLFormElement>;
+  onKeyDown: (event: React.KeyboardEvent<HTMLTextAreaElement>) => void;
+} {
+  const formRef = useRef<HTMLFormElement>(null);
+
+  const handleKeyDown = (
+    event: React.KeyboardEvent<HTMLTextAreaElement>,
+  ): void => {
+    if (
+      event.key === "Enter" &&
+      !event.shiftKey &&
+      !event.nativeEvent.isComposing
+    ) {
+      formRef.current?.requestSubmit();
+      event.preventDefault();
+    }
+  };
+
+  return { formRef, onKeyDown: handleKeyDown };
+}
+
+export default function ChatBot({
+  inputRef,
+  t,
+}: {
+  userName?: string;
+  inputRef?: React.Ref<HTMLTextAreaElement>;
+  t: TFunction;
+}) {
+  const {
+    messages,
+    input,
+    handleInputChange,
+    handleSubmit,
+    isLoading,
+    append,
+    reload,
+  } = useChat({
+    api: "/api/v0/chat",
+    initialMessages: [
+      { id: "-1", content: t("initial-message"), role: "assistant" },
+    ],
+  });
+  const { copyToClipboard, isCopied } = useCopyToClipboard({});
+  const { formRef, onKeyDown } = useEnterSubmit();
+  const messagesWrapperRef = useRef<HTMLDivElement>(null);
+
+  const userStyles = "rounded-br-none";
+  const botStyles = "rounded-bl-none";
+  const suggestions = [
+    {
```

Comment:
Would be awesome to have these translated in future if possible, curious to see how that works 😏

Reply:
These will be generated by the LLM in the future as well based on the MVP prompts for this.
So once we support different languages in the LLM we can also have the prompt suggestions in those languages :+1: